content/swift/concepts/protocols/terms/codingkey/codingkey.md

```diff
@@ -0,0 +1,160 @@
+```shell
+---
+Title: 'CodingKey'
+Description: 'Enables the mapping of JSON's keys to different Swift's struct / class properties.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Protocols'
+  - 'APIs'
+  - 'Enum'
+  - 'JSON'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+```
+
+**`CodingKey`** is an extension to the Swift model when parsing the JSON data object into a Swift object. It allows us to map JSON's data keys into different 
+property names in the Swift object.
+
+> **Note:** JSON and Swift have different naming conventions (JSON uses *snake_case* for its key identifiers, while Swift uses *camelCase* for its property identifiers)
+## Syntax
+This protocol will work when a Swift object conforms to either an `Decodable` or `Encodable` protocol.
```

Comment:
I don't believe there needs to be a separate example but `Decodable` and `Encodable` are each just half of the `Codable` protocol which isn't mentioned